src/client/Microsoft.Identity.Client/ApiConfig/AbstractConfidentialClientAcquireTokenParameterBuilder.cs

```diff
@@ -33,6 +36,54 @@ public override Task<AuthenticationResult> ExecuteAsync(CancellationToken cancel
         }
 
         internal IConfidentialClientApplicationExecutor ConfidentialClientApplicationExecutor { get; }
+
+#if DESKTOP || NET_CORE
+        /// <summary>
+        ///  Modifies the token acquisition request so that the acquired token is a Proof of Possession token (PoP), rather than a Bearer token. 
+        ///  PoP tokens are similar to Bearer tokens, but are bound to the HTTP request and to a cryptographic key, which MSAL can manage on Windows.
+        ///  See https://aka.ms/msal-net-pop
+        /// </summary>
+        /// <param name="httpRequestMessage">An HTTP request to the protected resource which requires a PoP token. The PoP token will be cryptographically bound to the request.</param>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item>This is an experimental API. The method signature may change in the future without involving a major version upgrade.</item>
+        /// <item> An Authentication header is automatically added to the request</item>
+        /// <item> The PoP token is bound to the HTTP request, more specifically to the HTTP method (GET, POST, etc.) and to the Uri (path and query, but not query parameters). </item>
+        /// <item> MSAL creates, reads and stores a key in memory that will be cycled every 8 hours.</item>
+        /// </list>
+        /// </remarks>
+        public T WithProofOfPosession(HttpRequestMessage httpRequestMessage)
```

Comment:
Since this method already exists on the public client, it would be better to add a single public API to the base class `AbstractAcquireTokenParameterBuilder`. This way PoP will be available on both public and confidential client.

Reply:
This class did not have access to the serviceBundle so I wasnt sure how to access it without dramatically changing the class. any thoughts

Reply:
Yeah, those objects are complex ... So your derived class has the ServiceBundle, then your base class should also be able to get a hold of it. In an inheritance hierarchy (e.g. Animal > Cat > PersianCat), the order in which each of the underlying objects are created is from base class to most derived class (i.e. Animal() gets build, then Cat(), then PersianCat()). However, you do have the ability to pass around params through each of these constructors, smth like: 

```csharp
        internal AbstractPublicClientAcquireTokenParameterBuilder(
            IPublicClientApplicationExecutor publicClientApplicationExecutor)
            : base(publicClientApplicationExecutor.ServiceBundle) // like this
        {
            PublicClientApplicationExecutor = publicClientApplicationExecutor;
        }

```

and in the base class:

```csharp
         internal IServiceBundle ServiceBundle { get; }

        internal AbstractAcquireTokenParameterBuilder(IServiceBundle serviceBundle)
        {
            ServiceBundle = serviceBundle;
        }
```

Reply:
This comment still stands.